src/librustdoc/html/layout.rs

```diff
@@ -194,9 +194,10 @@ pub fn render<T: fmt::Display, S: fmt::Display>(
                    .collect::<String>(),
     suffix=page.resource_suffix,
     extra_scripts=extra_scripts.iter().map(|e| {
-        format!("<script src=\"{root_path}{extra_script}\"></script>",
+        format!("<script src=\"{root_path}{extra_script}{suffix}.js\"></script>",
```

Comment:
This universally adds the resource suffix, but `source-files.js` is never saved with it, so it will never link correctly.